components/core/src/streaming_archive/writer/Archive.hpp

```diff
@@ -261,11 +275,13 @@ namespace streaming_archive { namespace writer {
 
         size_t m_target_segment_uncompressed_size;
         Segment m_segment_for_files_with_timestamps;
-        std::unordered_set<logtype_dictionary_id_t> m_logtype_ids_in_segment_for_files_with_timestamps;
-        std::unordered_set<variable_dictionary_id_t> m_var_ids_in_segment_for_files_with_timestamps;
+        ArrayBackedPosIntSet<logtype_dictionary_id_t> m_logtype_ids_in_segment_for_files_with_timestamps;
+        ArrayBackedPosIntSet<variable_dictionary_id_t> m_var_ids_in_segment_for_files_with_timestamps;
+        std::unordered_set<variable_dictionary_id_t> m_var_ids_without_timestamps_temp_holder;
```

Comment:
* How about a comment before saying `// Logtype and variable IDs for a file that hasn't yet been assigned to the timestamp or timestamp-less segment`?
* How about `m_var_ids_for_file_with_unassigned_segment` (and similarly for the logtype IDs)?
* For consistency, let's swap the order of these two sets (logtype then var).